### PR TITLE
Remove compact profile.

### PR DESCRIPTION
**Pull request by robertsipka**
The standard doesn't defines ECMAScript Compact Profile as a subset of Ecma-262 Edition 5.1.
Profile modes can be added easily like the minimal profile if required.

JerryScript-DCO-1.0-Signed-off-by: Robert Sipka rsipka.uszeged@partner.samsung.com

### CODE REVIEW

File: docs/04.INTERNALS.md [hunk 1]
```diff
@@ -247,9 +247,7 @@ Chunk size of the pool is 8 bytes (reduces fragmentation).
 ### Number
 
 There are two possible representation of numbers according to standard IEEE 754:
-
-* 4-byte (float, compact profile)
-* 8-byte (double, full profile)
+The default is 8-byte (double), but its able to use 4-byte (set CONFIG_ECMA_NUMBER_TYPE value), with the risk that it could lead inaccuracies.
```
Comment (zherczeg) [inline docs/04.INTERNALS.md]: split this into two lines.
but the engine supports the 4-byte (single precision) representation by setting CONFIG_ECMA_NUMBER_TYPE as well

File: jerry-core/ecma/builtin-objects/ecma-builtins.c [hunk 2]
```diff
@@ -807,20 +799,11 @@ ecma_builtin_dispatch_construct (ecma_object_t *obj_p, /**< built-in object */
...
 
   JERRY_ASSERT (!ecma_is_value_empty (ret_value));
 
   return ret_value;
 } /* ecma_builtin_dispatch_construct */
-
-/**
- * @}
- * @}
- */
```
Comment (zherczeg) [inline jerry-core/ecma/builtin-objects/ecma-builtins.c]: Please don't remove this comment.

File: jerry-core/ecma/operations/ecma-exceptions.c [hunk 3]
```diff
@@ -37,7 +37,8 @@
 ecma_object_t *
 ecma_new_standard_error (ecma_standard_error_t error_type) /**< native error type */
 {
-#ifndef CONFIG_ECMA_COMPACT_PROFILE_DISABLE_ERROR_BUILTINS
+
```
Comment (zherczeg) [inline jerry-core/ecma/operations/ecma-exceptions.c]: This newline is not needed

**Comment by zherczeg** [comment]
Good patch, LGTM after those stlye fixes.

**Comment by robertsipka** [comment]
Thanks, I've updated the patch based on your comments.

**Comment by LaszloLango** [comment]
LGTM

**Comment by zherczeg** [comment]
LGTM

**Comment by robertsipka** [comment]
Thanks, I've rebased it with master.

**Pull request merged**